mgmtworker/cloudify_system_workflows/tests/test_create_deployment.py

```diff
@@ -114,3 +114,52 @@ def test_labels_from_plan_and_kwargs(mock_ctx, mock_client, blueprint_plan):
     assert len(call.kwargs['labels']) == 2
     assert {'a': 'b'} in call.kwargs['labels']
     assert {'a': 'c'} in call.kwargs['labels']
+
+
+def test_inputs(mock_ctx, mock_client, blueprint_plan):
+    def parse_inputs(obj):
+        if type(obj) == dict:
+            for k, v in obj.items():
+                for constraint in v.get('constraints', []):
+                    if 'valid_values' in constraint and \
+                            type(constraint['valid_values']) == list:
+                        parsed_values = []
+                        for item in constraint['valid_values']:
+                            if item == {'get_secret': 's1'}:
+                                parsed_values.append('y')
+                            elif item == {'get_secret': 's2'}:
+                                parsed_values.append('z')
+                            else:
+                                parsed_values.append(item)
+                        constraint['valid_values'] = parsed_values
```

Comment:
I think this is WAY too much logic for a test util, and not enough assertions. What if obj isn't a dict, or the thing isn't a list? Well we don't really want to then return nothing, rather, we'd prefer to just fail an assert...
All in all, I think this function should looke more like,
```python
assert obj['a']['constraints']['valid_values'] == [...]
obj['a']['constraints']['valid_values'] = [hardcoded]
```
(+- more asserts as you see fit)

Anyway, the point is, I'd prefer test utils to be as straightforward as possible, and as strict as possible